### ISSUE TITLE
Update to newest segment java library (2.0.0-RC4)

### ISSUE THREAD

**Issue by JustinC474** [ardoq/analytics-clj#11]
To send integration specific data to segment, you need access to the `integrationOptions` function from the segment java library. Below is example code of how you would sending Amplitude a `session_id`:

```
analytics.enqueue(TrackMessage.builder("Clicked Button")
   .userId("f4ca124298")
   .context(ImmutableMap.builder()
       .put("ip", "12.212.12.49")
       .put("language", "en-us")
       .build()
   )
   .integrationOptions("Amplitude", ImmutableMap.of("session_id", "<Timestamp>"))
); 
```

However, this functionality is not available in 1.0.4. It is, however, available in 2.0.0.

I'm sure there is more functionality this Clojure wrapper does not have access to that would be gained by switching to the new Java library.

**Comment by bostonaholic** [comment]
@ebaxt Is this library being maintained any longer? We need support for Segment's 2.0 Java library. If not, we'll have to fork or create our own.

**Comment by ebaxt** [comment]
@bostonaholic Yes, we're still using it internally, but since we haven't had any need for 2.0 support yet, this haven't been prioritized. Would welcome a PR if you have time to contribute?

**Comment by rishimkumar** [comment]
@ebaxt we ended up needing it so we wrote a version.  Figured you might be interested: https://github.com/circleci/analytics-clj
